frontend/src/pages/Common/Budget.js

```diff
@@ -20,33 +20,43 @@ const styles = {
   cell: { display: "flex", justifyContent: "space-between" }
 };
 
-const renderProjectedBudgetAmount = (
+const renderProjectedBudgetAmount = ({
   projectedBudget,
   currIndex,
   editIndex,
   isEditing,
   budgetAmountEdit,
-  setBudgetAmountEdit
-) => {
+  setBudgetAmountEdit,
+  isValidBudgetAmountEdit,
+  setIsValidBudgetAmountEdit
+}) => {
   return (
     <TableCell align="right" data-test="saved-projected-budget-amount">
       {isEditing && editIndex === currIndex ? (
         <TextField
           label={strings.common.projected_budget}
           value={budgetAmountEdit}
-          onChange={e => setBudgetAmountEdit(e.target.value)}
+          onChange={e => {
+            if (/^[0-9,.-]*$/.test(e.target.value)) {
```

Comment:
explain regex or create own variable to describe what's the condition

Reply:
added var numberSignsRegex